### PR TITLE
Pack .cs files in DecisionTree.Sources as contentFiles

### PR DESCRIPTION
**Pull request by natemcmaster**
The "shared" sources packages will not be supported in VS2017. This upgrades Microsoft.AspNetCore.Routing.DecisionTree.Sources to use NuGet3 content files instead.

This is similar to what we did in https://github.com/aspnet/Common/pull/170.

cref https://github.com/aspnet/Coherence-Signed/issues/379

### CODE REVIEW

File: shared/DecisionTree/Microsoft.AspNetCore.Routing.DecisionTree.Sources.nuspec [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
+    <id>Microsoft.AspNetCore.Routing.DecisionTree.Sources</id>
+    <version>$version$</version>
+    <authors>Microsoft</authors>
+    <owners>Microsoft</owners>
```
Comment (pranavkm) [inline shared/DecisionTree/Microsoft.AspNetCore.Routing.DecisionTree.Sources.nuspec]: This isn't required.

File: shared/DecisionTree/Microsoft.AspNetCore.Routing.DecisionTree.Sources.nuspec [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>Components for building a DecisionTree.</description>
+    <tags></tags>
+    <serviceable>true</serviceable>
+    <contentFiles>
+       <files include="contentFiles/cs/**/*.cs" buildAction="Compile" />
+    </contentFiles>
+  </metadata>
+  <files>
+    <file src="**\*.cs" target="contentFiles/cs/netstandard1.0/" />
```
Comment (pranavkm) [inline shared/DecisionTree/Microsoft.AspNetCore.Routing.DecisionTree.Sources.nuspec]: Could we keep the slashes the same (forward slashes)? Different in the `contentFiles` node and here.

File: shared/DecisionTree/Microsoft.AspNetCore.Routing.DecisionTree.Sources.nuspec [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
+    <id>Microsoft.AspNetCore.Routing.DecisionTree.Sources</id>
+    <version>$version$</version>
+    <authors>Microsoft</authors>
+    <owners>Microsoft</owners>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>Components for building a DecisionTree.</description>
+    <tags></tags>
```
Comment (pranavkm) [inline shared/DecisionTree/Microsoft.AspNetCore.Routing.DecisionTree.Sources.nuspec]: Remove?

File: makefile.shade [hunk 4]
```diff
@@ -0,0 +1,28 @@
+
+var VersionPrefix = '1.2.0'
+var VERSION='0.1'
+var FULL_VERSION='0.1'
+use-standard-lifecycle
+k-standard-goals
+
+#pack-sources target='compile'
```
Comment (pranavkm) [inline makefile.shade]: Would it make sense to make this a part of KoreBuild? 

`if (Directory.Exists("shared") { Enumerate subdirs + pack }`

Comment (natemcmaster) [inline makefile.shade]: No, we're trying to minimalize usages of Sake going forward. MSBuild all things! (however begrudgingly).

**Comment by natemcmaster** [comment]
:watch: on https://github.com/aspnet/Universe/pull/466 to merge

File: makefile.shade (hunk 4 above)
Comment (natemcmaster) [inline makefile.shade]: ....and just realized my response makes no sense here. Was thinking of an almost identical change I made here: https://github.com/aspnet/Common/pull/179.

Yes, your suggestion make senses. We should add a better convention for .Sources packages.

**Pull request merged**